runtime/compiler/trj9/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -15832,31 +15768,116 @@ TR::Register *
 J9::Z::TreeEvaluator::BCDCHKEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    {
    PRINT_ME("BCDCHK", node, cg);
-   TR::Register* rv = node->getFirstChild()->getRegister();
+   TR::Node* pdopNode = node->getFirstChild();
+   TR::Register* resultReg = pdopNode->getRegister();
+   bool isResultPD = false, isVariableParam = false;
+   uint32_t firstCallParamIndex = 0;
+
+   static char* isVectorBCDEnv = feGetEnv("TR_enableVectorBCD");
+   bool isEnableVectorBCD = TR::Compiler->target.cpu.getS390SupportsVectorPackedDecimalFacility()
+                               && !TR::Options::getCmdLineOptions()->getOption(TR_DisableVectorBCD)
+                               || isVectorBCDEnv;
+
+   switch (pdopNode->getOpCodeValue())
+      {
+      case TR::pdcmpgt:
+      case TR::pdcmplt:
+      case TR::pdcmpge:
+      case TR::pdcmple:
+      case TR::pdcmpeq:
+      case TR::pdcmpne:
+      case TR::pd2l:
+      case TR::pd2i:
+      case TR::pd2iOverflow:
+      case TR::pd2lOverflow:
+         {
+         firstCallParamIndex = 1;
+         break;
+         }
+      case TR::i2pd:
+      case TR::l2pd:
+      case TR::pdshlOverflow:
+      case TR::pdshr:
+         {
+         firstCallParamIndex = 2;
+         isResultPD = true;
```

Comment:
Can't we use the type to determine this? i.e. `pdshl` is defined as:

```
   {
   /* .opcode               = */ TR::pdshr,
   /* .name                 = */ "pdshr",
   /* .properties1          = */ ILProp1::RightShift,
   /* .properties2          = */ ILProp2::ValueNumberShare | ILProp2::SupportedForPRE,
   /* .properties3          = */ ILProp3::LikeUse,
   /* .properties4          = */ ILProp4::BinaryCodedDecimalOp | ILProp4::CanHaveStorageReferenceHint,
   /* .dataType             = */ TR::PackedDecimal,
   /* .typeProperties       = */ ILTypeProp::PackedDecimal,
   /* .childProperties      = */ TWO_SAME_CHILD(TR::PackedDecimal),
   /* .swapChildrenOpCode   = */ TR::BadILOp,
   /* .reverseBranchOpCode  = */ TR::BadILOp,
   /* .booleanCompareOpCode = */ TR::BadILOp,
   /* .ifCompareOpCode      = */ TR::BadILOp,
   },
```

Type is specified as `TR::PackedDecimal` which is already stored in the node.